policy/query-builder/query-builder.go

```diff
@@ -0,0 +1,214 @@
+		tableAlias:      tableAlias,
+		randomGenerator: randomGenerator,
+	}
+}
+
+func (qb *BigQueryQueryBuilder) getAllowedParams(params map[string]string) []string {
+	paramsKeys := lib.GetMapKeys(params)
+	for _, value := range paramsHierarchy {
+		for k, v := range value {
+			if lib.SliceContains(paramsKeys, k) {
+				return v
+			}
+		}
+	}
+	return nil
+}
+
+func (qb *BigQueryQueryBuilder) filterParams(params map[string]string, allowedParams []string) map[string]string {
+	paramsKeys := lib.GetMapKeys(params)
+	for _, key := range paramsKeys {
+		if !lib.SliceContains(allowedParams, key) {
+			delete(params, key)
+		}
+	}
+	return params
+}
+
+func (qb *BigQueryQueryBuilder) processOrClauseParam(paramValue string) string {
+	whereClauses := make([]string, 0)
+	statusList := strings.Split(paramValue, ",")
+	for _, status := range statusList {
+		if val, ok := paramsWhereClause[status]; ok && val != "" {
+			whereClauses = append(whereClauses, val)
+		}
+	}
+	return "(" + strings.Join(whereClauses, " OR ") + ")"
+}
+
+func (qb *BigQueryQueryBuilder) processProducerUidParam(paramKey, paramValue string, queryParams map[string]interface{}) string {
+	tmp := make([]string, 0)
+	for _, uid := range strings.Split(paramValue, ",") {
+		randomIdentifier := qb.randomGenerator()
+		queryParams[randomIdentifier] = uid
+		tmp = append(tmp, fmt.Sprintf("'@%s'", randomIdentifier))
+	}
+	return fmt.Sprintf(paramsWhereClause[paramKey], strings.Join(tmp, ", "))
+}
+
+func (qb *BigQueryQueryBuilder) processParams(allowedParams []string, filteredParams map[string]string) ([]string, map[string]interface{}) {
+	whereClauses := make([]string, 0)
+	queryParams := make(map[string]interface{})
+
+	for _, paramKey := range allowedParams {
+		paramValue, exists := filteredParams[paramKey]
+		if !exists || paramValue == "" {
+			continue
+		}
+
+		if lib.SliceContains(orClausesKeys, paramKey) {
+			whereClause := qb.processOrClauseParam(filteredParams[paramKey])
+			whereClauses = append(whereClauses, whereClause)
+		} else if paramKey == "producerUid" {
+			whereClause := qb.processProducerUidParam(paramKey, paramValue, queryParams)
+			whereClauses = append(whereClauses, whereClause)
+		} else {
+			randomIdentifier := qb.randomGenerator()
+			whereClauses = append(whereClauses, fmt.Sprintf(paramsWhereClause[paramKey], randomIdentifier))
+			queryParams[randomIdentifier] = paramValue
+		}
+
+	}
+	return whereClauses, queryParams
+}
+
+func (qb *BigQueryQueryBuilder) BuildQuery(params map[string]string) (string, map[string]interface{}) {
+	const queryPrefix = "SELECT **tableAlias**.uid, **tableAlias**.name AS productName, " +
+		"**tableAlias**.codeCompany, CAST(**tableAlias**.proposalNumber AS INT64) AS proposalNumber, " +
+		"**tableAlias**.nameDesc,**tableAlias**.status, RTRIM(COALESCE(JSON_VALUE(**tableAlias**.data, " +
+		"'$.contractor.name'), '') || ' ' || " +
+		"COALESCE(JSON_VALUE(**tableAlias**.data, '$.contractor.surname'), '')) AS contractor, " +
```

Comment:
for the contractor:
I'm not sure if we should do the `RTRIM` + `COALESCE` combo here. Since the response model from the api could potentially change, it would have unnecessary impact in the query.
An alternative would be to keep `contractorName` and `contractorSurname` base on the values of `$.data.contractor.name/surname`, and when marshaling the response do the concatenation.

Reply:
we agreed to make this change later